src/main/java/com/ericsson/ei/handlers/EventToObjectMapHandler.java

```diff
@@ -81,30 +81,36 @@ public void setJmesPathInterface(JmesPathInterface jmesPathInterface) {
         return getEventToObjectList(eventId);
     }
 
+    /**
+     * To check and save the eventIds to the objectId in the mapped database.
+     * @param rulesObject
+     * @param event
+     * @param objectId aggregated event object Id
+     */
     public void updateEventToObjectMapInMemoryDB(RulesObject rulesObject, String event, String objectId) {
         String eventId = getEventId(rulesObject, event);
-        String condition = "{\"_id\" : \"" + eventId + "\"}";
-        ArrayList<String> list =  getEventToObjectList(eventId);
-        boolean firstTime = list.isEmpty();
-        list = updateList(list, eventId, objectId);
-        ObjectMapper mapper = new ObjectMapper();
-        JsonNode entry = null;
-
+        String condition = "{\"_id\" : \"" + objectId + "\"}";
+        LOGGER.debug("Checking document exists in the collection with condition : {}\n EventId : {}", condition, eventId);
+        boolean docExists = mongodbhandler.checkDocumentExists(databaseName, collectionName, condition);
         try {
-            entry = new ObjectMapper().readValue(condition, JsonNode.class);
-            ArrayNode jsonNode = mapper.convertValue(list, ArrayNode.class);
-            ((ObjectNode) entry).set(listPropertyName, mapper.readTree(jsonNode.toString()));
-            String mapStr = entry.toString();
-            LOGGER.debug("MongoDbHandler Insert/Update Event: {}\nto database: {} and to Collection: {}", mapStr, databaseName, collectionName);
-            if (firstTime) {
-                mongodbhandler.insertDocument(databaseName, collectionName, mapStr);
+        	if (!docExists) {
+        		ArrayList<String> list =  new ArrayList<String>();
+        		list.add(eventId);
+        		final ObjectMapper mapper = new ObjectMapper();
+    			JsonNode entry = new ObjectMapper().readValue(condition, JsonNode.class);
+        		ArrayNode jsonNode = mapper.convertValue(list, ArrayNode.class);
+        		((ObjectNode) entry).set(listPropertyName, mapper.readTree(jsonNode.toString()));
+                final String mapStr = entry.toString(); 
+            	LOGGER.debug("MongoDbHandler Insert/Update Event: {}\nto database: {} and to Collection: {}", mapStr, databaseName, collectionName);	
```

Comment:
I wonder if this is supposed to be logged on INFO level? But that can be handled later, if needed.

Reply:
yes we believe this can be kept as debug as long as we don't need them to be on the information level.